hatch/src/generators/tup/tuprules.rs

```diff
@@ -0,0 +1,128 @@
+use platform::arch::Arch;
+use project::Project;
+use project::ProjectKind;
+use project::Target;
+
+fn arch_flag(arch: &Arch) -> String {
+    match *arch {
+        Arch::X64 => String::from("-m64"),
+        Arch::X86 => String::from("-m32"),
+    }
+}
+
+fn type_flag(lib_type: &ProjectKind) -> String {
+    match *lib_type {
+        ProjectKind::Static => String::from("-static"),
+        ProjectKind::Shared => String::from("-dynamic"),
+        _ => String::from(""),
+    }
+}
+
+pub fn make_tuprules_string(project: &Project) -> String {
```

Comment:
a single public function. I love it. This one is particularly ugly, but atleast I know it is just a function. I think the above helpers could just be inlined here, but I left them as-is.